### PR TITLE
small changes 

### PR DESCRIPTION
**Pull request by iamAdrianIusca**
- removed static definition
- use emplace_back instead of push_back
- changed the c++ deprecated headers
- removed some redundant std containers initialization in the constructor
- removed redundant .clear call for the std containers in the destructor
- added some const reference for some std container parameters
- other small changes in different places

### CODE REVIEW

File: code/Common/Importer.h [hunk 1]
```diff
@@ -129,14 +129,8 @@ ImporterPimpl::ImporterPimpl() AI_NO_EXCEPT
 , mIsDefaultHandler( false )
 , mProgressHandler( nullptr )
 , mIsDefaultProgressHandler( false )
-, mImporter()
 , mPostProcessingSteps()
```
Comment (kimkulling) [inline code/Common/Importer.h]: Why?

File: code/Obj/ObjFileParser.cpp [hunk 2]
```diff
@@ -66,8 +66,7 @@ ObjFileParser::ObjFileParser()
 , m_pModel( nullptr )
 , m_uiLine( 0 )
 , m_pIO( nullptr )
-, m_progress( nullptr )
-, m_originalObjFileName() {
```
Comment (kimkulling) [inline code/Obj/ObjFileParser.cpp]: Why?

File: code/PostProcessing/ComputeUVMappingProcess.cpp [hunk 3]
```diff
@@ -51,10 +51,10 @@ using namespace Assimp;
 
 namespace {
 
-    const static aiVector3D base_axis_y(0.0,1.0,0.0);
```
Comment (kimkulling) [inline code/PostProcessing/ComputeUVMappingProcess.cpp]: What is the advantage?

File: code/PostProcessing/GenVertexNormalsProcess.cpp [hunk 4]
```diff
@@ -158,9 +158,9 @@ bool GenVertexNormalsProcess::GenMeshVertexNormals (aiMesh* pMesh, unsigned int
 
     // Set up a SpatialSort to quickly find all vertices close to a given position
     // check whether we can reuse the SpatialSort of a previous step.
-    SpatialSort* vertexFinder = NULL;
+    SpatialSort* vertexFinder = nullptr;
     SpatialSort  _vertexFinder;
-    ai_real posEpsilon = ai_real( 1e-5 );
```
Comment (kimkulling) [inline code/PostProcessing/GenVertexNormalsProcess.cpp]: We may use the max epsilon, but not initializing it is not a good idea ( was a finding from a static code analysis tool )

Comment (iamAdrianIusca) [reply code/PostProcessing/GenVertexNormalsProcess.cpp]: I have also some tool for this small things and that value is not used below.

File: code/PostProcessing/ComputeUVMappingProcess.cpp (hunk 3 above)
Comment (iamAdrianIusca) [reply code/PostProcessing/ComputeUVMappingProcess.cpp]: static in an anonymus namespace is redundant

File: code/Obj/ObjFileParser.cpp (hunk 2 above)
Comment (iamAdrianIusca) [reply code/Obj/ObjFileParser.cpp]: because the constructor for those fields it will be called anyway

File: code/Common/Importer.h (hunk 1 above)
Comment (iamAdrianIusca) [reply code/Common/Importer.h]: because the constructor for those fields it will be called anyway

**Comment by iamAdrianIusca** [comment]
@kimkulling let me know if I can provide more information

**Comment by iamAdrianIusca** [comment]
@kimkulling can you help me with this? please :)

**Comment by iamAdrianIusca** [comment]
@kimkulling I see some conflicts, please review the changes in order to know if worts the effort to resolve them.

**Comment by iamAdrianIusca** [comment]
Hi @kimkulling, can you help me with the windows build please? Thanks

**Comment by iamAdrianIusca** [comment]
@kimkulling I figure it out :) ready to be merged/rebased

**Comment by kimkulling** [comment]
Merge requests fixed at https://github.com/assimp/assimp/pull/3313.

**Pull request closed without merging**